Tests/MathLib/AutoCheckTools.h

```diff
@@ -67,6 +68,22 @@ struct IntervalTupleGenerator
     }
 };
 
+/// Generator for MxN fixed size eigen matrices with underlying type T.
+template <typename T, std::size_t M, std::size_t N, typename Gen = generator<T>>
+struct randomEigenMatrixGenerator
+{
+    Gen source;
+
+    using result_type = Eigen::Matrix<T, M, N>;
+
+    result_type operator()(std::size_t size = 0)
```

Comment:
It is a common pattern in the autocheck generators;
The autocheck is generating samples increasing the 'size' after every sample. So this argument can be used to generate e.g. longer and longer vectors/arrays/strings/magnitudes of numbers/etc. The 'size' passed to the `source` generator is "fixed", such that all the generated values are of the same magnitude.